internal/main.go

```diff
@@ -42,11 +42,14 @@ func main() {
 		stage        stages.Name
 		version      bool
 		logToStdout  bool
+
+		detectOfflineConfig string
 	}{}
 
 	flag.BoolVar(&flags.clearCache, "clear-cache", false, "clear any cached config")
 	flag.StringVar(&flags.configCache, "config-cache", "/run/ignition.json", "where to cache the config")
 	flag.DurationVar(&flags.fetchTimeout, "fetch-timeout", exec.DefaultFetchTimeout, "initial duration for which to wait for config")
+	flag.StringVar(&flags.detectOfflineConfig, "detect-config-provided", "", "If a config is provided, create a file at this path")
```

Comment:
The reason we write a stamp file is this is the best pattern I know of to signal the equivalent of `Result<bool>` as a subprocess.  Keying off distinct exit codes feels clunky since it breaks up the flow of using `set -e` in shell.